src/alfasim_score/converter/alfacase/convert_plugin_data.py

```diff
@@ -30,32 +30,62 @@ class ScoreAPBPluginConverter:
     def __init__(self, score_input_data: ScoreInputData):
         self.score_data = score_input_data
 
-    def _build_annular_temperature_table(self) -> AnnulusTemperatureTable:
+    def _build_annular_temperature_table(
+        self, final_temperature_depth: Scalar
+    ) -> AnnulusTemperatureTable:
         """
         Calculate the temperature of fluids based on temperature of formation coming from SCORE input file.
         It uses the well vertical positions in order to interpolates the temperatures of formation and it maps
         each measured depth to the temperature of formation in that position.
         """
-        measured_depths = Array(
+        formation_temperature_data = self.score_data.reader.read_formation_temperatures()
+        formation_depths = (
+            np.abs(formation_temperature_data["elevations"].GetValues())
+            + self.score_data.general_data["air_gap"].GetValue()
+        )
+        formation_temperatures = formation_temperature_data["temperatures"].GetValues()
+        trajectory = self.score_data.reader.read_well_trajectory()
+        formation_md_score_reference = Array(
+            np.interp(
+                formation_depths,
+                np.abs(trajectory["y"]),
+                np.abs(trajectory["md"]),
+            ),
+            LENGTH_UNIT,
+        )
+
+        formation_md_alfasim_reference = Array(
             [
-                self.score_data.get_position_in_well(depth).GetValue()
-                for depth in self.score_data.reader.read_well_trajectory()["md"]
+                self.score_data.get_position_in_well(Scalar(depth, LENGTH_UNIT)).GetValue()
+                for depth in formation_md_score_reference.GetValues(LENGTH_UNIT)
             ],
             LENGTH_UNIT,
         )
-        formation_temperature_data = self.score_data.reader.read_formation_temperatures()
-        trajectory = self.score_data.reader.read_well_trajectory()
-        interpolated_temperatures_y = Array(
+
+        annulus_depth_md_alfasim_reference = [
+            position
+            for depth in formation_md_score_reference.GetValues(LENGTH_UNIT)
+            for position in (
+                self.score_data.get_position_in_well(Scalar(depth, LENGTH_UNIT)).GetValue(),
+            )
+            if position < final_temperature_depth.GetValue()
+        ]
```

Comment:
This `self.score_data.get_position_in_well(Scalar(depth, LENGTH_UNIT)).GetValue()` gives you just one value, so the second loop is not necessary
```suggestion
        annulus_depth_md_alfasim_reference = [
            depth
            for depth in formation_md_score_reference.GetValues(LENGTH_UNIT)
            if self.score_data.get_position_in_well(Scalar(depth, LENGTH_UNIT)).GetValue() < final_temperature_depth.GetValue()
        ]
```

Reply:
By doing that, we won't be getting the final positions of the annuli, and we'll be passing exactly the positions of the formations:

```console
E       AssertionError: FILES DIFFER:
E       /tmp/pytest-of-noliveira/pytest-463/test_create_alfacase_file_scor0/test_convert_alfacase/test_create_alfacase_file_score_input_annulus_temp_table_.alfacase
E       /tmp/pytest-of-noliveira/pytest-463/test_create_alfacase_file_scor0/test_convert_alfacase/test_create_alfacase_file_score_input_annulus_temp_table_.obtained.alfacase
E       HTML DIFF: /tmp/pytest-of-noliveira/pytest-463/test_create_alfacase_file_scor0/test_convert_alfacase/test_create_alfacase_file_score_input_annulus_temp_table_.obtained.diff.html
E       --- 
E       +++ 
E       @@ -112,15 +112,13 @@
E                columns:
E                  temperature_depth_a:
E                    values:
E       -            - 0.0
E       -            - 1087.0
E       -            - 2788.0
E       +            - 2182.0
E       +            - 3269.0
E                    unit: m
E                  temperature_a:
E                    values:
E       -            - 4.0
E       -            - 47.0
E       -            - 61.90485113835376
E       +            - 56.59483362521891
E       +            - 69.3
E                    unit: degC
E              has_fluid_return_a: True
E              initial_leakoff_a:
E       @@ -160,13 +158,11 @@
E                columns:
E                  temperature_depth_b:
E                    values:
E       -            - 0.0
E       -            - 818.0
E       +            - 2182.0
E                    unit: m
E                  temperature_b:
E                    values:
E       -            - 4.0
E       -            - 36.358785648574056
E       +            - 56.59483362521891
E                    unit: degC
E              has_fluid_return_b: True
E              initial_leakoff_b:

```

Reply:
You can try
```suggestion
        annulus_depth_md_alfasim_reference = [
            position 
            for depth in formation_md_score_reference.GetValues(LENGTH_UNIT)
            if position := self.score_data.get_position_in_well(Scalar(depth, LENGTH_UNIT)).GetValue() < final_temperature_depth.GetValue()
        ]
```
If it does not work, then:
```suggestion
        annulus_depth_md_alfasim_reference = [
            self.score_data.get_position_in_well(Scalar(depth, LENGTH_UNIT)).GetValue() 
            for depth in formation_md_score_reference.GetValues(LENGTH_UNIT)
            if self.score_data.get_position_in_well(Scalar(depth, LENGTH_UNIT)).GetValue() < final_temperature_depth.GetValue()
        ]
```

Reply:
The second option works!